nucypher/blockchain/eth/actors.py

```diff
@@ -1544,6 +1547,19 @@ def block_until_ready(self, poll_rate: int = None, timeout: int = None):
                 self.nickname, self.nickname_metadata = nickname_from_seed(self.checksum_address)
                 break
 
+            # Provide periodic feedback to the user
```

Comment:
Feels like having two separate loops for bonding and funding would be simpler.

Reply:
Strictly speaking - its more computation, more loops, more code - simpler from what metric?

Reply:
In the sense of not requiring `if not bonded or not funded:` forks below, `funded` and `bonded` flags, and some conditions above.

Reply:
Two concurrent loops?

Reply:
Do they need to be concurrent? Looks like you can wait until bonded, and then wait until funded.

Reply:
Ah I see, Sure - I'm indifferent on that one but like a more simplistic way :-)

Reply:
@fjarri Feels like the current code is simple enough for now, but if we add another check then perhaps we could split it into a something like you mentioned - common `wait_for` type function with parameters to customize behaviour based on the check being performed